homeassistant/components/octoprint/sensor.py

```diff
@@ -124,11 +151,29 @@ def __init__(
     @property
     def native_value(self):
         """Return sensor state."""
+
+        # Get printer data from the coordinator
         printer: OctoprintPrinterInfo = self.coordinator.data["printer"]
         if not printer:
             return None
 
-        return printer.state.text
+        # Attempt to translate the state string from the API into an internal state value
+        api_state = printer.state.text
+        if value := _API_STATE_VALUE.get(api_state):
+            return value
+
+        # Got an unexpected state string from the API; fabricate a new internal state value and save it for next time
+        if match := _UNKNOWN_STATE_RE.match(api_state):
+            value = f"unknown_{match.group(1).lower().replace(' ', '_')}"
+        else:
+            value = api_state.lower().replace(" ", "_")
+
+        _LOGGER.warning(
+            "Unexpected Octoprint state string %r normalized to %r", api_state, value
+        )
+        _API_STATE_VALUE[api_state] = value
+        self._attr_options.append(value)
```

Comment:
A property should not have side effects like logging or adding an option to the options. I would argue that we should just return the states that we know and have mapped, and otherwise return `None`, which makes it show `unknown`. However, if there any benefit of logging an `Unknown State`, we can do that if we move this entity class over to set it's state via `_handle_coordinator_update` and `_attr_native_value`. I don't expect users to automate on an unknown state content. If they want to know it to diagnose their setup, we can implement `diagnostics.py` in the future so users can use that to download